### PR TITLE
New config parameters for serialization

### PR DESCRIPTION
**Pull request by Nek-**
This fix ticket #49.

The license of this code is MIT.

### CODE REVIEW

File: src/DependencyInjection/MisdPhoneNumberExtension.php [hunk 1]
```diff
@@ -39,6 +39,9 @@ public function load(array $configs, ContainerBuilder $container)
         }
         if ($config['serializer']['enabled']) {
             $loader->load('serializer.xml');
+
+            $container->setParameter('misd_phone_number.serializer.default_region', $config['serializer']['default_region']);
+            $container->setParameter('misd_phone_number.serializer.format', $config['serializer']['format']);
```
Comment (odolbeau) [inline src/DependencyInjection/MisdPhoneNumberExtension.php:44]: Clever!

File: src/Validator/Constraints/PhoneNumber.php [hunk 2]
```diff
@@ -45,7 +45,7 @@ class PhoneNumber extends Constraint
 
     public function getType(): ?string
     {
-        @trigger_error(__METHOD__.' is deprecated and will be removed in 4.0. Use "getTypes" instead.', E_USER_DEPRECATED);
+        @trigger_error(__METHOD__.' is deprecated and will be removed in 4.0. Use "getTypes" instead.', \E_USER_DEPRECATED);
```
Comment (odolbeau) [inline src/Validator/Constraints/PhoneNumber.php:48]: Previous PR to be merged first. :)

**Review by odolbeau — APPROVED** [review]
That's great! :ok_hand:

File: src/Validator/Constraints/PhoneNumber.php (hunk 2 above)
Comment (Nek-) [reply src/Validator/Constraints/PhoneNumber.php:48]: I fully agree (nice catch ;) ).

**Review by maxhelias — APPROVED** [review]
Great !

**Comment by maxhelias** [comment]
BTW we should also support the format on the validator

**Comment by Nek-** [comment]
@maxhelias I think having only 1 global configuration for default region and format has barely no benefit but adds a deprecation and limitation (if you want to decorrelate both you need to add custom code) that was not present before.

(as I stayed in the issue I originally agreed with what you just said)

**Comment by Nek-** [comment]
I added the demanded feature :) .

**Comment by Nek-** [comment]
Rebased :ok_hand:

**Comment by maxhelias** [comment]
Thanks @Nek- 😃

**Pull request merged**